plugin-support.patch

```diff
@@ -40,6 +40,52 @@ index 1a7535447cb..aad058ff30f 100644
    /// \brief Output warnings from headers matching this filter. Warnings from
    /// main files will always be displayed.
    llvm::Optional<std::string> HeaderFilterRegex;
+diff --git a/tools/extra/clang-tidy/tool/CMakeLists.txt b/tools/extra/clang-tidy/tool/CMakeLists.txt
+index f58cfea5518..1f9ebc0c3b1 100644
+--- a/tools/extra/clang-tidy/tool/CMakeLists.txt
++++ b/tools/extra/clang-tidy/tool/CMakeLists.txt
+@@ -5,13 +5,14 @@ set(LLVM_LINK_COMPONENTS
+   support
+   )
+ 
+-add_clang_tool(clang-tidy
++set(LLVM_OPTIONAL_SOURCES "dummy.cpp")
++add_clang_library(clangTidyMain
+   ClangTidyMain.cpp
+   )
+-add_dependencies(clang-tidy
++add_dependencies(clangTidyMain
+   clang-headers
+   )
+-target_link_libraries(clang-tidy
++target_link_libraries(clangTidyMain
+   PRIVATE
+   clangAST
+   clangASTMatchers
+@@ -39,11 +40,22 @@ target_link_libraries(clang-tidy
+   )
+ 
+ if(CLANG_ENABLE_STATIC_ANALYZER)
+-  target_link_libraries(clang-tidy PRIVATE
++  target_link_libraries(clangTidyMain PRIVATE
+     clangTidyMPIModule
+   )
+ endif()
+ 
++# supress warning about unknown source file
++set(LLVM_OPTIONAL_SOURCES "ClangTidyMain.cpp")
```

Comment:
ClangTidyMain.cpp is the main file of clang-tidy. It's not included for clang-tidy here because clang-tidy now links against clangTidyMain, which already includes ClangTidyMain.o.